src/ui/views/Table/TableView.svelte

```diff
@@ -102,6 +108,76 @@
       },
     });
   }
+
+  function handleColumnAppend() {
+    new CreateFieldModal($app, fields, async (field, value) => {
+      await api.addField(field, value);
+
+      buttonEl.scrollIntoView({
+        block: "nearest",
+        inline: "nearest",
+        behavior: "smooth",
+      });
+
+      updateFieldCfg(field);
+      updateViewCfg(field);
+    }).open();
+  }
+
+  function handleColumnInsert(anchor: string, direction: number) {
+    new CreateFieldModal($app, fields, async (field, value) => {
+      const position = fields.findIndex((f) => anchor === f.name) + direction;
+      await api.addField(field, value, position);
+
+      updateFieldCfg(field);
+      updateViewCfg(field, position);
```

Comment:
`updateFieldCfg` would update project-level field config, while `updateViewCfg` will overwrite the stale config in `project` with a newer `config`. Same reason mentioned in `onColumnConfigure`